### PR TITLE
Improve saving of user progress as the user types

### PR DESCRIPTION
**Pull request by MaksymShokin**
@tonytlwu @squallstar 

## Issue
Fliplet/fliplet-studio#1686

## Description
Added function that saves progress on input.

## Screenshots/screencasts
save demo

## Backward compatibility
This change is fully backward compatible.

## Notes
If the user changes screen too fast after typing (approximately less then 1s), it might not save. Apparently there is nothing we can do in that case.

### CODE REVIEW

File: js/libs/core.js [hunk 1]
```diff
@@ -86,7 +86,11 @@ Fliplet.FormBuilder = (function() {
             $($vm.$el).removeClass('has-error');
           }
         }
-      }
+      };
+
+      component.methods.saveProgress = function($event) {
+        this.$emit('_input', this.name, $event.target.value);
```
Comment (tonytlwu) [inline js/libs/core.js]: @squallstar This isn't debounced, while the user could be typing very quickly, firing this many times. Do you see a problem with it?

File: js/libs/core.js [hunk 2]
```diff
@@ -86,7 +86,11 @@ Fliplet.FormBuilder = (function() {
             $($vm.$el).removeClass('has-error');
           }
         }
-      }
+      };
+
+      component.methods.saveProgress = function($event) {
```
Comment (squallstar) [inline js/libs/core.js]: Please rename this method to `onInput`

File: js/libs/core.js (hunk 1 above)
Comment (squallstar) [reply js/libs/core.js]: I agree with debouncing this to something like 200ms. @tonytlwu confirm a good debounce time in regards to average typing speed :-)

File: templates/components/email.build.hbs [hunk 3]
```diff
@@ -3,9 +3,10 @@
   class="form-control"
   v-model.trim.lazy="value"
   v-on:blur="updateValue()"
+  v-on:input="saveProgress"
```
Comment (squallstar) [inline templates/components/email.build.hbs]: `v-on:input="onInput()"`

File: js/libs/core.js (hunk 1 above)
Comment (MaksymShokin) [reply js/libs/core.js]: Fixed

File: js/libs/core.js (hunk 2 above)
Comment (MaksymShokin) [reply js/libs/core.js]: Fixed

File: templates/components/email.build.hbs (hunk 3 above)
Comment (MaksymShokin) [reply templates/components/email.build.hbs]: Fixed

**Pull request merged**